Update the release note with the standard lib changes (#163)
ballerina-platform/ballerina-release: release-notes/swan-lake-alpha5-release-note.md

#### New Features

#### Improvements

> `jballerina.java` Package

Reviewer: Shouldn't this be
```suggestion
##### `jballerina.java` Package
```

or
```suggestion
##### Java Package
```

> 

Reviewer: ```suggestion
- Dependently-typed `java:cast` Function

```

> Improved the `cast` function to determine return type based on the inferred type.

Reviewer: ```suggestion
The `java:cast` function is now a dependently-typed function. If the `typedesc` argument is not provided, it is inferred from the contextually-expected type.
```

```bal

> FileInputStream|error obj = java:cast(inputStream);

Reviewer: ```suggestion
FileInputStream|error obj1 = java:cast(inputStream, FileInputStream);
FileInputStream|error obj2 = java:cast(inputStream); // The second argument is inferred to be `FileInputStream`.
```

#### New Features

#### Improvements
##### `jballerina.java` Package Updates
- Dependently-typed `java:cast` Function

The `java:cast` function is now a dependently-typed function. If the `typedesc` argument is not provided, it is inferred from the contextually-expected type.
```bal
FileInputStream|error obj1 = java:cast(inputStream, FileInputStream);
FileInputStream|error obj2 = java:cast(inputStream); // The second argument is inferred to be `FileInputStream`.
```

##### File Package Updates

> - The `path` field in the `ListenerConfig` has changed from optional to mandatory.

Reviewer 2: ```suggestion
- Changed the `path` field in the `ListenerConfig` from optional to mandatory.
```

- Improved the compiler plugin validation for the service.

##### Log Package Updates

> - Changed the time format in log output to RFC3339

Reviewer 2: ```suggestion
- Changed the time format in the log output to RFC3339.
```

> - Added validation to check log levels in Config.toml file

Reviewer 2: ```suggestion
- Added validation to check log levels in the `Config.toml` file.
```

> ##### Sql Package Updates

Reviewer 2: ```suggestion
##### SQL Package Updates
```

> If return types of the SQL time related data types are expected as records, then the type of those records are now validating and should be as follows, 

Reviewer 2: ```suggestion
If return types of the SQL time-related data types are expected as records, then the type of those records will be validated now and should be as follows.
```

- DATE -> time:Date
- TIME -> time:TimeOfDay
- TIME WITH TIME ZONE -> time:TimeOfDay
- TIMESTAMP -> time:Civil
- TIMESTAMP WITH TIME ZONE -> time:Civil

#### Bug Fixes
##### Log Package Bug fixes

> Made log module escape characters properly (https://github.com/ballerina-platform/ballerina-standard-library/issues/1192).

Reviewer 2: ```suggestion
Made log module escape characters properly (https://github.com/ballerina-platform/ballerina-standard-library/issues/1192).
```
It is not clear what has been done.

Author: @daneshk

Reviewer 3: Shall we change like `Invalid Characters in log message are escaped properly`

Reviewer 2: @IrushiL - Please change to "Invalid characters are now escaped properly in log messages."

Author: Done
